### DISCUSSION TITLE
Gateway Time-out

### DISCUSSION THREAD

**Discussion by yakman000** [Budibase/budibase#8551] [Help]
Dear All,
after successful installation of budibase, on the same machine, I installed iRedMail and RoundCube to manage the necessary email server with the accounts to budibase login.
After a machine reboot, I'm not able to login into budibase :(
at the https://x.x.x.x.:10000/builder/auth/login I have "Gateway Time-out" :-/
How can i fix this ?
I've already developed apps into this installation, in the worst case how can i download those to another budibase installation ?

Best Regards

**Comment by melohagan** [comment]
Hey @yakman000 

Do you see any errors in the in **bbapps** or **bbworker** container logs? 
Are either of those email server ports being used?

Have you allocated enough resources in docker?

**Reply by yakman000** [reply]
Hi @melohagan 
I found the container log of bbapps and bbworker, you may find both in attach.
[bbworker.log.txt](https://github.com/Budibase/budibase/files/9971165/bbworker.log.txt)
[bbapps.log.txt](https://github.com/Budibase/budibase/files/9971166/bbapps.log.txt)

**Comment by yakman000** [comment]
Hey @melohagan 
The iredadmin (email server use 7791 port)

I installed budibase using docker compose following this page instructions: https://docs.budibase.com/docs/docker-compose on Rocky Linux 9.

Please,
May you point me out where I can find the logs of bbapps, bbworker ?
I'm not familiar with docker: How can I manage the resources allocated to docker ? What could be a reasonable value/values ?

Thank you in advance to your support.

**Comment by melohagan** [comment]
Hey @yakman000 

The way I increase docker resources is to install docker desktop, click on the cog icon, then go to resources:
<img width="795" alt="Screenshot 2022-11-08 at 17 29 02" src="https://user-images.githubusercontent.com/101575380/200634306-dfcbae84-e36c-4b53-8f20-5d2a58960ef5.png">

**This is way more than needed if all you're running is Budibase**
<img width="971" alt="Screenshot 2022-11-08 at 17 29 18" src="https://user-images.githubusercontent.com/101575380/200634361-42472823-a2c5-4368-8ff8-8a16b701844d.png">

**Reply by melohagan** [reply]
https://docs.budibase.com/docs/docker#recommended-resources

**Reply by yakman000** [reply]
Hi @melohagan 
![docker resources](https://user-images.githubusercontent.com/15886165/200804559-a34b7621-4655-4692-9170-f4fba6b17e22.jpg)

these is the output of "docker stats", is it ok for you?

**Reply by melohagan** [reply]
Seeing the logs is easy with docker desktop. You can simply click into the container and you'll see the logs:

<img width="341" alt="Screenshot 2022-11-09 at 20 10 39" src="https://user-images.githubusercontent.com/101575380/200931158-d591c54a-98ba-4557-a57b-95b00f85a367.png">

<img width="454" alt="Screenshot 2022-11-09 at 20 11 15" src="https://user-images.githubusercontent.com/101575380/200931294-c220bcdf-3c37-4ac7-8296-62672023ea77.png">

**Reply by yakman000** [reply]
Thank you @melohagan
below you may find the requested log file. I found there various errors, but I haven't figure out the problem. Please, may you help me?

[bbworker.log.txt](https://github.com/Budibase/budibase/files/9979492/bbworker.log.txt)
[bbapps.log.txt](https://github.com/Budibase/budibase/files/9979493/bbapps.log.txt)

**Reply by melohagan** [reply]
Hey @yakman000 Thanks for those. Seems to be a few errors going on here.

I did see some 409 conflicts, so first thing that may be worth trying is to see if there are any conflicts in your couchdb instance and resolve them.
Guide here: https://github.com/Budibase/budibase/discussions/7777#discussioncomment-3732390

**Reply by yakman000** [reply]
Hey @melohagan 

accessing using the web browser to <url>:4005/_utils gave me "Unable to connect"
Using this notes : https://docs.budibase.com/docs/accessing-couchdb I tried : <url>/db/_utils/# this gave me "404 Not Found" nginx
so I checked again if the couchdb container runs: docker ps
![docker resources2](https://user-images.githubusercontent.com/15886165/201070546-ee3098d8-1e05-488a-b3a9-1c7e6b8d0ee7.jpg)

:-\ ..I'm not able to access the couchdb web page, Where is my mistake about this?

**Reply by melohagan** [reply]
Hey @yakman000 

You'll need to port forward and restart with a `docker-compose down` `docker-compose up`.

Add this to your `docker-compose.yaml`
```yaml
couchdb-service:
    ports: 
      - "${COUCH_DB_PORT}:5984"
```

I'll update the docs.

**Reply by yakman000** [reply]
> Hey @yakman000 Thanks for those. Seems to be a few errors going on here.
> 
> I did see some 409 conflicts, so first thing that may be worth trying is to see if there are any conflicts in your couchdb instance and resolve them. Guide here: [#7777 (comment)](https://github.com/Budibase/budibase/discussions/7777#
![fauxton hang](https://user-images.githubusercontent.com/15886165/201349212-35fb4b6d-87ff-41b8-9275-dc9edb34758d.jpg)
discussioncomment-3732390)


Following your guide I have fixed someone couchdb conflicts, but now the fauxton hang and seems as the couchdb doesn't works.

Therefore I shutdown the containers, and later try to keep these up, but also this operation made troubles.
![badShutdown](https://user-images.githubusercontent.com/15886165/201367144-35b4dbc5-3744-44b0-a3f7-e04d35789d7b.jpg)
